worktree_status.go

```diff
@@ -266,6 +266,19 @@ func (w *Worktree) Add(path string) (plumbing.Hash, error) {
 	return h, err
 }
 
+func (w *Worktree) removeIfDeleted(path string, hash plumbing.Hash, err error) (plumbing.Hash, error) {
+	pathError, ok := err.(*os.PathError)
```

Comment:
you don't need to do the cast or check if is a lsstat, the function is smart enough for detect any non-existan file

Reply:
Also I rather having the if condition where the `err` is generated instead of having another function. Thanks

Reply:
Done.